src/Compiler/TypedTree/TypedTree.fs

```diff
@@ -4565,6 +4565,13 @@ type DecisionTreeCase =
     member x.DebugText = x.ToString()
 
     override x.ToString() = sprintf "DecisionTreeCase(...)"
+    
+[<Struct; NoComparison; RequireQualifiedAccess>]
+type ActivePatternReturnKind =
+    | RefTypeWrapper
+    | StructTypeWrapper
+    | Boolean
+    member this.IsStruct with get () = this <> RefTypeWrapper 
```

Comment:
It is likely that `<>` will be calling an `Equals` which will do boxing. We should do something like
```fsharp
     member this.IsStruct with get () = match this with RefTypeWrapper -> true | _ -> false
```
So it will only do the tag comparison.

Reply:
In case of matching, we can also add coequality to the type.

Reply:
Vlad is correct, see codegen difference here:
https://sharplab.io/#v2:DYLgZgzgNALiCWwA+BtAPAZRgJwK4GMYBuAAgDkB7AYQoFsAHAQ23ggoDtSAlAUwEdc8bDwCKuRsHhh4PACYBBfPh4QIAPgC6AWABQMAJ70eJRTHgA3HgAVGMGD2zteMXI4DS8drJIBeXSQCApBJeMAAVQx4AdWxGeiNsf0CSYKw8QgijGLiEpMDggCEKCmAeRnY8gNoeWgAjBxIYAAtWADoASQg0ghgAUQAPekl8eBgSAHdRppIAcx4xgAoASl8SWlt8aebWCamUkJ5wyOz4hoBaNRIwCQhjYIB9EgvGvB5KtZr67EaWiA6unA9ADyLBmngku2as3mJGWq22EBIaEuoUy0Vip2+QA==


( going via `<>` boxes and calls Equals, explicit matching will do a comparison of the numeric tag)

Reply:
fixed